### DISCUSSION TITLE
Question about Using Supabase for Edge Functions, Authentication, Database, and Cloudflare R2 for File Storage

### DISCUSSION THREAD

**Discussion by mehrdad-tat** [supabase/supabase#17480] [Questions]
Can I use [Supabase ](https://supabase.com/)for authentication, database, Edge Functions, and [Cloudflare R2](https://www.cloudflare.com/developer-platform/r2/) for file storage in my project? 

I want to make sure that during file uploads and downloads, there won't be any associated costs with Supabase. Also, is it possible for files to be directly uploaded to [R2 ](https://www.cloudflare.com/developer-platform/r2/)and downloaded from [R2 ](https://www.cloudflare.com/developer-platform/r2/)without routing through Supabase?

**Comment by vigneshksaithal** [comment]
Yeah, you can definitely use it. I am also doing the same.
For reference read this https://www.okupter.com/blog/upload-files-cloudflare-r2-in-sveltekit
